src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -300,10 +300,37 @@ internal unsafe struct MethodTable
         private uint Flags;
         [FieldOffset(4)]
         public uint BaseSize;
+        [FieldOffset(0x0e)]
+        public ushort InterfaceCount;
+        [FieldOffset(ParentMethodTableOffset)]
+        public MethodTable* ParentMethodTable;
+        [FieldOffset(InterfaceMapOffset)]
+        public nuint* InterfaceMap;
```

Comment:
```suggestion
        public MethodTable** InterfaceMap;
```

(This should simplify some of the casts.)